src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/OffenceCodeLookupService.kt

```diff
@@ -0,0 +1,179 @@
+    val paragraph9 = OffenceCodeParagraph("9", "Administers a controlled drug to themself or fails to prevent the administration of a controlled drug to them by another person (but subject to rule 52 - defences to rule 51(9))")
+    val paragraph10 = OffenceCodeParagraph("10", "Is intoxicated as a consequence of knowingly consuming any alcoholic beverage")
+    val paragraph11 = OffenceCodeParagraph("11", "Knowingly consumes any alcoholic beverage other than that provided to them pursuant to a written order under rule 25 (alcohol and tobacco) (1)")
+    val paragraph12 = OffenceCodeParagraph("12", "Has in his possession either (a) any unauthorised article, or (b) a greater quantity of any article than he is authorised to have")
+    val paragraph13 = OffenceCodeParagraph("13", "Sells or delivers to any person any unauthorised article")
+    val paragraph14 = OffenceCodeParagraph("14", "Sells or, without permission, delivers to any person any article which they are allowed to have only for their own use")
+    val paragraph15 = OffenceCodeParagraph("15", "Takes improperly any article belonging to another person or to a prison")
+    val paragraph16 = OffenceCodeParagraph("16", "Intentionally or recklessly sets fire to any part of a prison or any other property, whether or not their own")
+    val paragraph17 = OffenceCodeParagraph("17", "Destroys or damages any part of a prison or any other property, other than their own")
+    val paragraph18 = OffenceCodeParagraph("18", "Absents himself from any place he is required to be or is present at any place where they are not authorised to be")
+    val paragraph19 = OffenceCodeParagraph("19", "Is disrespectful to any officer, or any person (other than a prisoner) who is at the prison for the purpose of working there, or any person visiting a prison")
+    val paragraph20 = OffenceCodeParagraph("20", "Uses threatening, abusive or insulting words or behaviour")
+    val paragraph21 = OffenceCodeParagraph("21", "Intentionally fails to work properly or, being required to work, refuses to do so")
+    val paragraph22 = OffenceCodeParagraph("22", "Disobeys any lawful order")
+    val paragraph23 = OffenceCodeParagraph("23", "Disobeys or fails to comply with any rule or regulation applying to them")
+    val paragraph24 = OffenceCodeParagraph("24", "Receives any controlled drug, or, without the consent of an officer, any other article, during the course of a visit (not being an interview such as is mentioned in rule 38, which relates to visits from legal advisors)")
+    val paragraph24a = OffenceCodeParagraph("24a", "Displays, attaches or draws on any part of a prison, or on any other property, threatening, abusive or insulting racist words, drawings, symbols or other material")
+
+    return OffenceCodeLookupBuilder()
+      .code(0).nomisCodes("1A", "1B").notCommittedOnOwnNomisCodes("25D").paragraph(paragraph1)
```

Comment:
I need to know your offence codes to do this

Reply:
Yeah - sorry - I now have write access to the spread sheet but need to make some change to add a bunch of rows - cause currently we have some rows which end like 'if this was x then y' - should be easy enough to deal with - but want to check before I go ahead and do it